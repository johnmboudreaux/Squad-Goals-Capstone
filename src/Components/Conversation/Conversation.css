.smack-talk h1 {
  color: #282828;
  font-weight: bold;
  font-size: 2em;
  line-height: 2em;
  margin: 0;
  height: 50px; }

.comment-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: #d2d3d3;
  border: #282828 10px solid;
  height: 250px;
  overflow: scroll; }
