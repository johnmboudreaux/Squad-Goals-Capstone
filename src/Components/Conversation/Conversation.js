import React from 'react';
import PropTypes from 'prop-types';
import Comment from '../Comment/Comment';

const Conversation = ({ comments }) => {
  const commentCards = comments.map((comment) => {
    return (
      <Comment commentData={comment} key={`comment-${comment.id}`} />
    );
  });

  return (
    <div className="smack-talk">
      <h1>Smack Talk Board</h1>
      { commentCards }
    </div>
  );
};

Conversation.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.object),
};

export default Conversation;
