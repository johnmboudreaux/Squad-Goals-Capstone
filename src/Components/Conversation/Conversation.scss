@import '../../variables.scss';


.smack-talk h1 {
  color: $black;
  font-weight: bold;
  font-size: 2em;
  line-height: 2em;
  margin: 0;
  height: 50px;
}

.comment-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: $white;
  border: $black 10px solid;
  height: auto;
  overflow: scroll;
}
