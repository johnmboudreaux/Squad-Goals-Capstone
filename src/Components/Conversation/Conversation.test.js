/* eslint-disable */
import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Conversation from './Conversation';
import conversationData from '../../../data/test/comments_data_test';

describe('MainApp snapshot', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<Conversation
      comments={conversationData}
    />);

    expect(wrapper).toMatchSnapshot();
  });


  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Conversation
      comments={conversationData}
    />, div);
  });
});
