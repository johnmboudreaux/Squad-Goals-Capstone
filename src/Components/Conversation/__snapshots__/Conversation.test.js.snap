// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Conversation
    comments={
      Array [
        Object {
          "body": "this is the comments body",
          "conversation_id": 1,
          "id": 1,
        },
        Object {
          "body": "this is the comments body 2",
          "conversation_id": 1,
          "id": 2,
        },
        Object {
          "body": "this is the comments body 3",
          "conversation_id": 3,
          "id": 3,
        },
        Object {
          "body": "this is the comments body 4",
          "conversation_id": 4,
          "id": 4,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Smack Talk Board
        </h1>,
        <div
          className="comment-container"
        >
          <Comment
            commentData={
              Object {
                "body": "this is the comments body",
                "conversation_id": 1,
                "id": 1,
              }
            }
          />
          <Comment
            commentData={
              Object {
                "body": "this is the comments body 2",
                "conversation_id": 1,
                "id": 2,
              }
            }
          />
          <Comment
            commentData={
              Object {
                "body": "this is the comments body 3",
                "conversation_id": 3,
                "id": 3,
              }
            }
          />
          <Comment
            commentData={
              Object {
                "body": "this is the comments body 4",
                "conversation_id": 4,
                "id": 4,
              }
            }
          />
        </div>,
      ],
      "className": "smack-talk",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Smack Talk Board",
        },
        "ref": null,
        "rendered": "Smack Talk Board",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Comment
              commentData={
                Object {
                  "body": "this is the comments body",
                  "conversation_id": 1,
                  "id": 1,
                }
              }
            />,
            <Comment
              commentData={
                Object {
                  "body": "this is the comments body 2",
                  "conversation_id": 1,
                  "id": 2,
                }
              }
            />,
            <Comment
              commentData={
                Object {
                  "body": "this is the comments body 3",
                  "conversation_id": 3,
                  "id": 3,
                }
              }
            />,
            <Comment
              commentData={
                Object {
                  "body": "this is the comments body 4",
                  "conversation_id": 4,
                  "id": 4,
                }
              }
            />,
          ],
          "className": "comment-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "comment-1",
            "nodeType": "function",
            "props": Object {
              "commentData": Object {
                "body": "this is the comments body",
                "conversation_id": 1,
                "id": 1,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "comment-2",
            "nodeType": "function",
            "props": Object {
              "commentData": Object {
                "body": "this is the comments body 2",
                "conversation_id": 1,
                "id": 2,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "comment-3",
            "nodeType": "function",
            "props": Object {
              "commentData": Object {
                "body": "this is the comments body 3",
                "conversation_id": 3,
                "id": 3,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "comment-4",
            "nodeType": "function",
            "props": Object {
              "commentData": Object {
                "body": "this is the comments body 4",
                "conversation_id": 4,
                "id": 4,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Smack Talk Board
          </h1>,
          <div
            className="comment-container"
          >
            <Comment
              commentData={
                Object {
                  "body": "this is the comments body",
                  "conversation_id": 1,
                  "id": 1,
                }
              }
            />
            <Comment
              commentData={
                Object {
                  "body": "this is the comments body 2",
                  "conversation_id": 1,
                  "id": 2,
                }
              }
            />
            <Comment
              commentData={
                Object {
                  "body": "this is the comments body 3",
                  "conversation_id": 3,
                  "id": 3,
                }
              }
            />
            <Comment
              commentData={
                Object {
                  "body": "this is the comments body 4",
                  "conversation_id": 4,
                  "id": 4,
                }
              }
            />
          </div>,
        ],
        "className": "smack-talk",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Smack Talk Board",
          },
          "ref": null,
          "rendered": "Smack Talk Board",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Comment
                commentData={
                  Object {
                    "body": "this is the comments body",
                    "conversation_id": 1,
                    "id": 1,
                  }
                }
              />,
              <Comment
                commentData={
                  Object {
                    "body": "this is the comments body 2",
                    "conversation_id": 1,
                    "id": 2,
                  }
                }
              />,
              <Comment
                commentData={
                  Object {
                    "body": "this is the comments body 3",
                    "conversation_id": 3,
                    "id": 3,
                  }
                }
              />,
              <Comment
                commentData={
                  Object {
                    "body": "this is the comments body 4",
                    "conversation_id": 4,
                    "id": 4,
                  }
                }
              />,
            ],
            "className": "comment-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "comment-1",
              "nodeType": "function",
              "props": Object {
                "commentData": Object {
                  "body": "this is the comments body",
                  "conversation_id": 1,
                  "id": 1,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "comment-2",
              "nodeType": "function",
              "props": Object {
                "commentData": Object {
                  "body": "this is the comments body 2",
                  "conversation_id": 1,
                  "id": 2,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "comment-3",
              "nodeType": "function",
              "props": Object {
                "commentData": Object {
                  "body": "this is the comments body 3",
                  "conversation_id": 3,
                  "id": 3,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "comment-4",
              "nodeType": "function",
              "props": Object {
                "commentData": Object {
                  "body": "this is the comments body 4",
                  "conversation_id": 4,
                  "id": 4,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
